README.md

```diff
@@ -1,43 +1,419 @@
-cloud-sdk-docker
-================
 
-> :warning: **IMPORTANT NOTE:** The base image for the `cloud-sdk` image will be changed to Alpine Linux on or after 5/15/17.
-> For more information, please see: [#58](https://github.com/GoogleCloudPlatform/cloud-sdk-docker/issues/58)
+# Google Cloud SDK Docker
+
+
+The [Cloud SDK](https://cloud.google.com/sdk/) is the command-line interface to the Google Cloud Platform.  It is a flexible utility which uses GCPs own Cloud APIs to perform many different tasks such as deploying code on AppEngine, to creating Compute Engine VMs to checking IAM permissions and so on.  It is the command line interface to pretty much every Google Cloud Platform API and service.   
+
+This article describes some more ways customers can use it by demonstrating several use cases involving containerizing the Cloud SDK.   The following shows Dockerfiles targeted towards various operating systems and some sample usecases (Its nothing new, just a sample set to document the baseline dockerfiles.
+
+* [Dockerfile](#dockerfile)
+    - [Ubuntu](#ubuntu)
+    - [CentOS](#centos)
+    - [Alpine](#alpine)
+* [Usecases](#usecases)
+    - [Containerize local development environment](#containerize-local-development-environment)
+    - [Run gcloud cli without installing SDK locally](#run-gcloud-cli-without-installing-sdk-locally)
+    - [Run emulators in containers](#run-emulators-in-containers)
+    - [Automate simple DevOps tasklets](#automate-simple-devdps-tasklets)       
+
+## Dockerfile
+
+To use any of the images below, enter the folder for the OS in question and run
+
+```bash
+docker build -t cloud-sdk .
+```
+
+If you want to use a prebuilt **Alpine** based image, see
+
+* [https://hub.docker.com/r/google/cloud-sdk/](https://hub.docker.com/r/google/cloud-sdk/)
+
+specifically
+
+```bash
+docker run -ti google/cloud-sdk:latest
+```
+
+or via tagged version of the SDK:
+
+```bash
+docker run -ti google/cloud-sdk:153.0.0
+docker run -ti google/cloud-sdk:152.0.0
+docker run -ti google/cloud-sdk:151.0.1
+```
+
+>> **NOTE:** with any of these images, you can also install the optional components and add that to your baseline container
+```
+$ docker run -ti google/cloud-sdk:latest gcloud components list
+
+Your current Cloud SDK version is: 153.0.0
+The latest available version is: 153.0.0
+
+┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
```

Comment:
This is an exhaustive list that's subject to change all the time. It's a too long output to show in the README.md. Remember, README.md files should be brief for others to actually go through. It's not exhaustive documentation. So consider making your point more brief here.

In the **NOTE** you say you can install components, then you show a command that doesn't do that.